Date: Fri Mar  4 22:39:57 CET 2005
From: Vojtech Pavlik <vojtech@suse.cz>
Subject: Fix psmouse.resetafter

=====================================================================================

ChangeSet@1.2073, 2005-03-04 22:38:24+01:00, vojtech@suse.cz
  input: Make psmouse.resetafter also trigger on time-based sync
         loss detection, not just in the data-based case.
  
  Signed-off-by: Vojtech Pavlik <vojtech@suse.cz>


 psmouse-base.c |    6 ++++++
 1 files changed, 6 insertions(+)


diff -Nru a/drivers/input/mouse/psmouse-base.c b/drivers/input/mouse/psmouse-base.c
--- a/drivers/input/mouse/psmouse-base.c	2005-03-04 22:41:10 +01:00
+++ b/drivers/input/mouse/psmouse-base.c	2005-03-04 22:41:10 +01:00
@@ -175,6 +175,12 @@
 		printk(KERN_WARNING "psmouse.c: %s at %s lost synchronization, throwing %d bytes away.\n",
 		       psmouse->name, psmouse->phys, psmouse->pktcnt);
 		psmouse->pktcnt = 0;
+
+		if (psmouse->resetafter) {
+			psmouse->state = PSMOUSE_IGNORE;
+			printk(KERN_NOTICE "psmouse.c: issuing reconnect request\n");
+			serio_reconnect(psmouse->ps2dev.serio);
+		}
 	}
 
 	psmouse->last = jiffies;
